+++ /usr/lib/python2.6/site-packages/nova/virt/libvirt/config.py
@@ -894,6 +894,31 @@
 
         return dev
 
+class LibvirtConfigGuestVideo(LibvirtConfigGuestDevice):
+
+    def __init__(self, **kwargs):
+        super(LibvirtConfigGuestVideo, self).__init__(root_name="video",
+                                                      **kwargs)
+
+        self.type = 'cirrus'
+        self.vram = None
+        self.heads = None
+
+    def format_dom(self):
+        dev = super(LibvirtConfigGuestVideo, self).format_dom()
+
+        model = etree.Element("model")
+        model.set("type", self.type)
+
+        if self.vram:
+            model.set("vram", str(self.vram))
+
+        if self.heads:
+            model.set("heads", str(self.heads))
+
+        dev.append(model)
+
+        return dev
 
 class LibvirtConfigGuestHostdev(LibvirtConfigGuestDevice):
     def __init__(self, **kwargs):
